src/v/datalake/partitioning_writer.cc

```diff
@@ -30,6 +30,17 @@ operator<<(std::ostream& o, const partitioning_writer::partitioned_file& f) {
     return o;
 }
 
+ss::future<> partitioning_writer::flush() {
```

Comment:
nit: I do understand that this method returns an exception as a convenience resulting from the fact that we make calls inside the function passed to `max_concurrent_for_each` which can't return result. However, the method declaration is a inconsistent with others which return a result. We should contain this inconsistency by catching the error and converting it to a result.